include/rdma/providers/fi_log.h

```diff
@@ -70,7 +70,11 @@ int fi_log_enabled(const struct fi_provider *prov, enum fi_log_level level,
 		   enum fi_log_subsys subsys);
 void fi_log(const struct fi_provider *prov, enum fi_log_level level,
 	    enum fi_log_subsys subsys, const char *func, int line,
-	    const char *fmt, ...);
+	    const char *fmt, ...)
+#if defined(__clang__) || defined(__GNUC__)
+	__attribute__ ((__format__ (__printf__, 6, 7)));
+#endif
```

Comment:
This is an external exposed header file.  Let's get agreement on whether this should be added to an exported call.  (It seems fine).

For the definition, please do something like:
```
#if defined (__GNUC__) ...
#define ATTRIBUTE_PRINTF(x, y) __attribute__ ((__format__(__printf__, x, y)
#else
#define ATTRIBUTE_PRINTF(x, y)
#endif

fi_log(...)  ATTRIBUTE_PRINTF(6, 7);
```
So, we can get rid of the weird #ifdef around the semi-colon and keep things readable.

Reply:
@jsquyres @j-xiong @a-ilango - please note proposed change to exported header file (provider fi_log() call).

Reply:
Seems like the ifdef isn't needed at all because:

    include/windows/osd.h:#define __attribute__(x)

Reply:
The changes look good. 👍